### PR TITLE
Remove Symbian support

### PR DESCRIPTION
**Pull request by ilmari**
Symbian never made it beyond being a toy port, and the platform is long obsolete.

This also lets us get rid of the `PERL_GLOBAL_STRUCT` compilation option and its associated `dVAR`,
which people kept forgetting to add to functions that needed it, as well as `USE_HEAP_INSTEAD_OF_STACK` and
`SETSOCKOPT_OPTION_VALUE_T`.

### CODE REVIEW

File: locale.c [hunk 1]
```diff
@@ -4947,64 +4947,6 @@ Perl__is_cur_LC_category_utf8(pTHX_ int category)
...
-                DEBUG_Lv(PerlIO_printf(Perl_debug_log, "\t?error messages for %s are UTF-8=%d\n",
-                                    save_input_locale,
-                                    is_utf8));
-                goto finish_and_return;
-            }
-
-            DEBUG_L(PerlIO_printf(Perl_debug_log, "All error messages for %s contain only ASCII; can't use for determining if UTF-8 locale\n", save_input_locale));
-        }
-
-#    endif
```
Comment (xenu) [inline locale.c]: This seems to be unrelated to `PERL_GLOBAL_STRUCT`.

**Review by xenu — APPROVED** [review]

File: locale.c (hunk 1 above)
Comment (ilmari) [reply locale.c]: Yeah, that was `unifdef` being abit too eager because of the `#if 0`. Hunk reverted.

**Comment by smpeters** [comment]
I love the smell of chainsaw in the morning.
